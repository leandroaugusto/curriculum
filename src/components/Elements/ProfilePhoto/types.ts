export interface ProfilePhotoProps {
  url?: string;
}
