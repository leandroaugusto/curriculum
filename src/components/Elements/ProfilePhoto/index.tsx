import Photo2 from "@/assets/images/facebook.jpeg";
import Image from "next/image";

import { ProfilePhotoProps } from "./types";

export default function ProfilePhoto({
  url,
}: ProfilePhotoProps): React.ReactElement {
  return (
    <Image
      src={url || Photo2}
      alt="Profile photo"
      className="object-cover object-center h-32 w-32 md:h-40 md:w-40 rounded-full"
    />
  );
}
