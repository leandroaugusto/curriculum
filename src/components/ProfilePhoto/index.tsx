import Photo from "@/assets/images/photo.jpeg";
import Image from "next/image";

export default function ProfilePhoto() {
  return (
    <Image
      src={Photo}
      alt="Profile photo"
      className="object-cover object-center h-40 w-40 rounded-full"
    />
  );
}
