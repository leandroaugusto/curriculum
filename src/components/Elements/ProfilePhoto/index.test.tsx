import { render } from "@testing-library/react";
import ProfilePhoto from ".";
import { faker } from "@faker-js/faker";

jest.mock("@/assets/images/facebook.jpeg");

describe("ProfilePhoto", () => {
  it("renders the ProfilePhoto element", () => {
    const image_url = faker.image.url();
    const { container } = render(<ProfilePhoto url={image_url} />);

    expect(container).toMatchSnapshot();
  });

  it("renders the ProfilePhoto without url prop", () => {
    const { container } = render(<ProfilePhoto />);

    expect(container).toMatchSnapshot();
  });
});
