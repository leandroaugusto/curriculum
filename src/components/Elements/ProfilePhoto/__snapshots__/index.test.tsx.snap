// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePhoto renders the ProfilePhoto element 1`] = `
<div>
  <img
    alt="Profile photo"
    class="object-cover object-center h-32 w-32 md:h-40 md:w-40 rounded-full"
    data-nimg="1"
    decoding="async"
    height="100"
    loading="lazy"
    src="/_next/image?url=https%3A%2F%2Floremflickr.com%2F640%2F480%3Flock%3D4995502768652288&w=256&q=75"
    srcset="/_next/image?url=https%3A%2F%2Floremflickr.com%2F640%2F480%3Flock%3D4995502768652288&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Floremflickr.com%2F640%2F480%3Flock%3D4995502768652288&w=256&q=75 2x"
    style="color: transparent;"
    width="100"
  />
</div>
`;

exports[`ProfilePhoto renders the ProfilePhoto without url prop 1`] = `
<div>
  <img
    alt="Profile photo"
    class="object-cover object-center h-32 w-32 md:h-40 md:w-40 rounded-full"
    data-nimg="1"
    decoding="async"
    height="100"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
    style="color: transparent;"
    width="100"
  />
</div>
`;
